src/pybind/mgr/rbd_support/module.py

```diff
@@ -41,32 +47,55 @@
 
 REPORT_MAX_RESULTS = 64
 
-
-class Module(MgrModule):
-    COMMANDS = [
-        {
-            "cmd": "rbd perf image stats "
-                   "name=pool_spec,type=CephString,req=false "
-                   "name=sort_by,type=CephChoices,strings="
-                   "write_ops|write_bytes|write_latency|"
-                   "read_ops|read_bytes|read_latency,"
-                   "req=false ",
-            "desc": "Retrieve current RBD IO performance stats",
-            "perm": "r"
-        },
-        {
-            "cmd": "rbd perf image counters "
-                   "name=pool_spec,type=CephString,req=false "
-                   "name=sort_by,type=CephChoices,strings="
-                   "write_ops|write_bytes|write_latency|"
-                   "read_ops|read_bytes|read_latency,"
-                   "req=false ",
-            "desc": "Retrieve current RBD IO performance counters",
-            "perm": "r"
-        }
-    ]
-    MODULE_OPTIONS = []
-
+RBD_TASK_OID = "rbd_task"
+
+TASK_ACTION_FLATTEN = "flatten"
+TASK_ACTION_REMOVE = "remove"
+TASK_ACTION_TRASH_REMOVE = "trash remove"
+TASK_ACTION_MIGRATION_EXECUTE = "migrate execute"
+TASK_ACTION_MIGRATION_COMMIT = "migrate commit"
+TASK_ACTION_MIGRATION_ABORT = "migrate abort"
+VALID_TASK_ACTIONS = [TASK_ACTION_FLATTEN,
+                      TASK_ACTION_REMOVE,
+                      TASK_ACTION_TRASH_REMOVE,
+                      TASK_ACTION_MIGRATION_EXECUTE,
+                      TASK_ACTION_MIGRATION_COMMIT,
+                      TASK_ACTION_MIGRATION_ABORT]
+
+TASK_SEQUENCE = "sequence"
+TASK_ID = "id"
+TASK_ACTION = "action"
+TASK_REFS = "refs"
+TASK_MESSAGE = "message"
+TASK_RETRY_TIME = "retry_time"
+TASK_IN_PROGRESS = "in_progress"
+TASK_PROGRESS = "progress"
+TASK_CANCELED = "canceled"
+
+TASK_REF_POOL_NAME = "pool_name"
+TASK_REF_POOL_NAMESPACE = "pool_namespace"
+TASK_REF_IMAGE_NAME = "image_name"
+TASK_REF_IMAGE_ID = "image_id"
+
+TASK_RETRY_INTERVAL = timedelta(seconds=30)
+
+
+def extract_pool_key(pool_spec):
+    if not pool_spec:
+        return GLOBAL_POOL_KEY
+
+    match = re.match(r'^([^/]+)(?:/([^/]+))?$', pool_spec or '')
```

Comment:
nit: why do you need `or ''` here if you check for `not pool_spec` above?

Reply:
I think it was left-over from an earlier implementation.